### PR TITLE
API updates

### PR DESCRIPTION
**Pull request by jcampbell**
What's here:
* Minor build script tweak for deployment convenience
* Fix for multiple log entries during load of PandasDataset batches
* Refactor normalize_data_asset_name to no longer be considered an internal method
* Updates to create_expectations notebook to clarify get_batch api
* netlify index page

### CODE REVIEW

File: great_expectations/init_notebooks/create_expectations.ipynb [hunk 1]
```diff
@@ -35,14 +35,26 @@
...
+     "evalue": "name 'ge' is not defined",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
+      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
+      "\u001b[0;32m<ipython-input-1-0dd8d48bfd61>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcontext\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mge\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdata_context\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataContext\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
+      "\u001b[0;31mNameError\u001b[0m: name 'ge' is not defined"
+     ]
+    }
+   ],
```
Comment (Aylr) [inline great_expectations/init_notebooks/create_expectations.ipynb]:
```suggestion
"outputs": [],
```

File: great_expectations/init_notebooks/create_expectations.ipynb [hunk 2]
```diff
@@ -69,7 +81,25 @@
...
+    "Your data_asset_name will consist of three components: a datasource, generator, and generator_asset, but you can usually provide only some of them as long as the name is not ambiguous. See [more in the reference](https://docs.great_expectations.io/en/latest/reference/data_context_reference.html#data-asset-names). "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_asset_name = \"npidata_pfile\" # TODO: replace with your value!\n",
```
Comment (Aylr) [inline great_expectations/init_notebooks/create_expectations.ipynb]: remove the npi bit

**Comment by jcampbell** [comment]
Note: this PR also changes the datasource and generator return types for getting available generator_asset and partition_id names to be list instead of set (to make it easier to add sorting later)

File: great_expectations/init_notebooks/create_expectations.ipynb (hunk 2 above)
Comment (jcampbell) [inline great_expectations/init_notebooks/create_expectations.ipynb]:
```suggestion
    "data_asset_name = \"REPLACE_ME\"  # TODO: replace with your value!\n",
```

File: great_expectations/init_notebooks/create_expectations.ipynb [hunk 3]
```diff
@@ -107,51 +136,28 @@
...
-    "\n",
-    "* A. To validate an existing table:\n",
-    "\n",
-    "```python\n",
-    "data_asset_name = 'CHOOSE THE NAME OF YOUR TABLE FROM THE LIST OF DATA ASSETS ABOVE'\n",
-    "batch = context.get_batch(data_asset_name, \n",
-    "                        expectation_suite_name='my_suite'\n",
-    "                        BatchKwargs(table=data_asset_name)) \n",
-    "```\n",
+    "BatchKwargs define the specific batch of data a datasource should fetch. Depending on the datasource, batch kwargs might include a filepath, table name, SQL query, or even an existing DataFrame loaded outside of Great Expectations. Choose the best batch_kwargs for your situation."
```
Comment (Aylr) [inline great_expectations/init_notebooks/create_expectations.ipynb]:
```suggestion
    "How you get a batch of data will depend on your datasource (pandas, SQL, Spark, etc). `BatchKwargs` tell your specific datasource how to fetch a batch of data and might include a filepath, table name, SQL query, or even an existing DataFrame loaded outside of Great Expectations. Choose the best batch_kwargs for your situation."
```

File: great_expectations/init_notebooks/create_expectations.ipynb [hunk 4]
```diff
@@ -182,7 +187,19 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# The datasource can add and store additional identifying information to ensure you can track a batch through\n",
+    "# your pipeline\n",
```
Comment (Aylr) [inline great_expectations/init_notebooks/create_expectations.ipynb:190]: these would be even better in a markdown cell above this cell.

**Pull request merged**